superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -125,7 +125,9 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
     if (dashboardId) {
       try {
         const result = await this.loadDashboard(dashboardId);
-        if (result) {
+        if (
+          result?.owners.some((owner: any) => owner.id === this.props.userId)
```

Comment:
It may break if `owners` is `undefined` or `null` at this point, correct?

Reply:
nit: should we have a method `isOwner(owners: any[])` that does this computation and in the condition we just call it? so the condition inside the `if` is a bit more simple to read and to extend if needed later on?

Reply:
> It may break if `owners` is `undefined` or `null` at this point, correct?

I guess, but `owners` should never have either of these values, its empty state with is an empty array.

> nit: should we have a method isOwner(owners: any[]) that does this computation and in the condition we just call it? so the condition inside the if is a bit more simple to read and to extend if needed later on?

Yeah, I can make that change

Reply:
> Why do we need this check? Shouldn't `loadDashboard` return `null` if the user does not have access to the dashboard? Keep in mind that frontend code can always be modified and checks bypassed.

As stated in the PR description, this check is to prevent the dashboard dropdown on this modal from pre-populating if the user is not an owner/cannot edit the dashboard.  It's possible that the user has access to view the dashboard, but not edit it.

Reply:
I mean that `loadDashboard` has a filter where user is the owner, so it will return `null` otherwise.

Reply:
the plural `loadDashboards` used to populate the async select does, but the singular `loadDashboard` used here does not; it uses a different endpoint that I don't believe supports the filter that `loadDashboards` uses

Reply:
@jfrag1 You're right. Unfortunately the single version does not support filters 🤦🏼‍♂️  

Maybe we could use the `/api/v1/dashboard` like this:

```
{
  "columns": [
    "id",
    "dashboard_title"
  ],
  "filters": [
    {
      "col": "id",
      "opr": "eq",
      "value": <dashboardId>
    },
    {
      "col": "owners",
      "opr": "rel_m_m",
      "value": <userId>
    }
  ]
}
```

Reply:
Just to explain a little bit more. My concern is that security checks should always happen at the server side because it's really simple to edit the frontend scripts and bypass this types of checks.

Reply:
Nevermind. Someone could also edit the query from the client side. We would need to change the `/api/v1/dashboard/<id>` endpoint to actually implement this correctly 😢

Reply:
I agree that security checks should always happen server-side, but I wouldn't classify the check done here as a security check.  It's more of a check that prevents the user from trying to do something that would be stopped from the actual security check on the backend.

For example, if we take this PR with only the backend changes, the frontend would still pre-populate, but saving the chart would fail with the non-owned dashboard pre-selected.

Reply:
True, but only if you already have access to view that dashboard/its metadata anyways.  Nothing is being leaked that the user can't access normally; the `GET api/v1/dashboard/<pk>` endpoint has its own security and will 404 if the user shouldn't have access to view it

Reply:
Good point 👍🏼